File: simple_example.scala
import java.util.{Random, Date, LinkedHashMap}
import java.lang.{String, Double, System}
import com.espertech.esper.client

class Listener extends client.UpdateListener {
    override def update(newData :Array[client.EventBean], oldData :Array[client.EventBean]) {
        val evb = newData(0)
        printf("Symbol: %s Price: %5.2f  Ts: %s\n", evb.get("symbol"),
            evb.get("price"), evb.get("ts"))
    }
}

object SimpleExample {

    val gen = new Random()
    def random_tick() = {
        val hmm = new LinkedHashMap[String, Object]()
        hmm.put("symbol", "AAPL")
        val p = new Double(gen.nextInt(18))
        hmm.put("price", p)
        hmm.put("ts", new Date(System.currentTimeMillis()))
        hmm
    }

    def main(args: Array[String]) {
        var type_map = new LinkedHashMap[String, Object]()
        type_map.put("symbol", classOf[String])
        type_map.put("price", classOf[Double])
        type_map.put("ts", classOf[Date])
        val conf = new client.Configuration()
        conf.addEventType("StockTick", type_map)
        val cep = client.EPServiceProviderManager.getProvider("myCEPEngine", conf)
        val cepRT = cep.getEPRuntime()
        val cepAdm = cep.getEPAdministrator()
        val stmt ="SELECT * FROM StockTick(symbol='AAPL').win:length(10) having avg(price) > 10.0"
        val cepStatement = cepAdm.createEPL(stmt)
        val h = new Listener()
        cepStatement.addListener(h)
        for (i <- 0 to 100) {
            var t = random_tick()
            cepRT.sendEvent(t, "StockTick")
        }
    }
}
SimpleExample.main(null)

File: simple_example.py
# Simple Jython example inspired by:
# http://coffeeonesugar.wordpress.com/2009/07/21/getting-started-with-esper-in-5-minutes/
from java.util import Random, Date, LinkedHashMap
from java.lang import String, Double, System
import com.espertech.esper.client as C

type_map = LinkedHashMap()
type_map.put("symbol", String)
type_map.put("price", Double)
type_map.put("ts", Date)

gen = Random()

def random_tick():
    hmm = LinkedHashMap()
    hmm.put("symbol", 'AAPL')
    p = Double(gen.nextInt(18))
    hmm.put("price", p)
    hmm.put("ts", Date(System.currentTimeMillis()))
    return hmm

class Listener(C.UpdateListener):
    def update(*args, **kwargs):
        a = args[1][0]
        print "Symbol: %s Price: %5.2f  Ts: %s" % (a.get("symbol"),
            a.get("price"), a.get("ts"))

def main():
    conf = C.Configuration()
    conf.addEventType("StockTick", type_map)
    cep = C.EPServiceProviderManager.getProvider("myCEPEngine", conf)
    cepRT = cep.getEPRuntime()
    cepAdm = cep.getEPAdministrator()
    stmt ="SELECT * FROM StockTick(symbol='AAPL').win:length(10) having avg(price) > 10.0"
    cepStatement = cepAdm.createEPL(stmt)
    h = Listener()
    cepStatement.addListener(h)
    for i in range(100):
        t = random_tick()
        cepRT.sendEvent(t, "StockTick")

if __name__ == '__main__':
    main()
